### ISSUE TITLE
CoordSysBuilder fails on anonymous dimensions

### ISSUE THREAD

**Issue by lesserwhirls** [Unidata/netcdf-java#228]
One way to trigger the bug is to open `cdmUnitTest/formats/grib2/GFS_Global_2p5deg_20101023_0600.grib2` (or `https://thredds.ucar.edu/thredds/dodsC/grib/NCEP/GFS/CONUS_80km/Best`) using `NetcdfDatasets`. What happens is that in the `ucar.nc2.internal.dataset.CoordinatesHelper` `isComplete()` method, we turn a list of dimension names into an immutable list:

https://github.com/Unidata/netcdf-java/blob/67953c00df446cec2db7870ee226881a15532274/cdm/core/src/main/java/ucar/nc2/internal/dataset/CoordinatesHelper.java#L216

but an anonymous dimension does not have a name (is `null`), and `com.google.common.collect.ImmutableSet.copyOf` does not like that:

~~~java
java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:878)
	at com.google.common.collect.ImmutableSet$Builder.add(ImmutableSet.java:510)
	at com.google.common.collect.ImmutableSet$Builder.add(ImmutableSet.java:475)
	at com.google.common.collect.ImmutableCollection$Builder.addAll(ImmutableCollection.java:456)
	at com.google.common.collect.ImmutableSet$Builder.addAll(ImmutableSet.java:541)
	at com.google.common.collect.ImmutableSet.copyOf(ImmutableSet.java:275)
	at ucar.nc2.internal.dataset.CoordinatesHelper$Builder.isComplete(CoordinatesHelper.java:216)
	at ucar.nc2.internal.dataset.CoordSystemBuilder.makeCoordinateSystemsMaximal(CoordSystemBuilder.java:536)
	at ucar.nc2.internal.dataset.CoordSystemBuilder.buildCoordinateSystems(CoordSystemBuilder.java:213)
	at ucar.nc2.internal.dataset.DatasetEnhancer.enhance(DatasetEnhancer.java:125)
	at ucar.nc2.dataset.NetcdfDatasets.enhance(NetcdfDatasets.java:179)
	at ucar.nc2.dataset.NetcdfDatasets.openDataset(NetcdfDatasets.java:151)
~~~

**Comment by tombrauer** [comment]
I recently hit this bug with netcdfAll-5.3.3 when trying to open this grib file: 
[ST4.2005010100.01h.zip](https://github.com/Unidata/netcdf-java/files/5330658/ST4.2005010100.01h.zip)

The file was retrieved from: https://data.eol.ucar.edu/cgi-bin/codiac/fgr_form/id=21.093

I was able to open the file using v4.6.15 but noticed the regression after migrating to v5.3.*

**Comment by lesserwhirls** [comment]
I believe this has been fixed in `5.4.0-SNAPSHOT` - if you'd like to give that a try, it can be accessed [here](https://artifacts.unidata.ucar.edu/repository/unidata-all/edu/ucar/netcdfAll/5.4.0-SNAPSHOT/netcdfAll-5.4.0-20201003.065226-16.jar).

**Comment by tombrauer** [comment]
I updated my build script to use cdm-core and grib libraries v5.4.0-SNAPSHOT and can confirm this bug is fixed. Thanks!
